components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -93,49 +120,76 @@ private AuthenticatorAdapterConstants.UserType resolveIdpType() {
                 AuthenticatorAdapterConstants.UserType.LOCAL : AuthenticatorAdapterConstants.UserType.FEDERATED;
     }
 
-    private User resolveLocalUserFromUserStore(AuthenticationContext context, AuthenticatedUserData.UserStore userStore)
-            throws ActionExecutionResponseProcessorException {
-
-        AbstractUserStoreManager userStoreManager = resolveUserStoreManager(context, userStore.getName());
-        try {
-            return userStoreManager.getUser(authenticatedUser.getUserId(), null);
-
-        } catch (org.wso2.carbon.user.core.UserStoreException e) {
-            String message = "Error occurred when trying to resolve local user by user Id:" + user.getUser().getId();
-            throw new ActionExecutionResponseProcessorException(message, e);
+    private String resolveUserIdFromResponse() throws ActionExecutionResponseProcessorException {
 
-        } catch (UserIdNotFoundException e) {
-            String message = "No user found for the given user Id:" + user.getUser().getId();
-            throw new ActionExecutionResponseProcessorException(message, e);
+        if (StringUtils.isNotBlank(userFromResponse.getUser().getId())) {
+            return userFromResponse.getUser().getId();
         }
+        // Todo: Add diagnostic log for the error scenario.
+        throw new ActionExecutionResponseProcessorException("The 'userId' field is missing in the authentication " +
+                "action response.");
     }
 
-    private void resolveFederatedUser() {
+    private AuthenticatedUserData.UserStore resolveUserStoreForLocalUser() {
 
-        authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
-                user.getUser().getId());
-        authenticatedUser.setFederatedIdPName(context.getExternalIdP().getIdPName());
-        authenticatedUser.setFederatedUser(true);
+        if (userFromResponse.getUser().getUserStore() != null) {
+            return userFromResponse.getUser().getUserStore();
+        }
+        // Todo: Add diagnostic log for the scenario.
+        return null;
```

Comment:
Can we return PRIMARY from here itself and load based on this later. This gives clarity

Reply:
The default userStore for product is `PRIMARY` and for asgardeo its is `DEFAULT`.